### ISSUE TITLE
"es2020.string" - Property 'matchAll' does not exist on type 'string'.ts(2339)

### ISSUE THREAD

**Issue by tommck** [microsoft/TypeScript#35129]
No matter the target/lib settings I set, I cannot get `string.matchAll` to be recognized by the editor tooling anywhere, including the Playground.

**TypeScript 3.7.2**
[Playground link](https://www.typescriptlang.org/play/?target=99#code/MYewdgzgLgBAZiEMC8MDkALAlmg3AKHwHoiYBbAQymAwEEAbemLCGAEwFM4swO3mwMAEQAmAAziAdNABOPAOZCANDGKkA7thrNWPYPQCunfgCMAnsPHjlqkjE1ZtLAfqN8Y54Rwi8AHlCF8UEhYEwoZFHhESUpqOkYACiJJIgBKIA)

Compiler Options:
```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

**Input:**
```typescript
const foo = 'hi';

// matchAll is defined in "2020.string", 
// which is included by "2020", 
// which is included by "esnext"
const bar = foo.matchAll(/./)
```

**Output:**
```javascript
"use strict";
const foo = 'hi';
// matchAll is defined in "2020.string", 
// which is included by "2020", 
// which is included by "esnext"
const bar = foo.matchAll(/./);

```

**Expected behavior:**
`matchAll` should be recognized and it's not.
Even the playground link for #32737 doesn't work

**Comment by tommck** [comment]
I think this is due to a bad assumption on my part and some weird problems locally.. closing this

**Comment by nsaunders** [comment]
Did you have a solution for this, by chance?

**Comment by nsaunders** [comment]
Solved. Here's my tsconfig.json:
```
{
  "compilerOptions": {
    "target": "esnext"
  }
}
```

I was making a noob mistake: I did not realize that `tsconfig.json` is ignored when running `tsc` on specific files.

**Comment by heavenkiller2018** [comment]
@nsaunders  i do it follow you. but the problem is still existed!

**Comment by Sjiep** [comment]
@heavenkiller2018 I bumped into the same issue, apparently it is only included in typescript version 3.7 and higher. See also #32737
